source/AI.cpp

```diff
@@ -1244,6 +1244,20 @@ bool AI::Stop(Ship &ship, Command &command, double maxSpeed)
 		double reverseTime = (180. - degreesToTurn) / ship.TurnRate();
 		reverseTime += speed / reverseAcceleration;
 		
+		// If you want to end up facing a specific direction, add the extra turning time.
+		if(direction)
+		{
+			// Time to turn from facing backwards to target:
+            double degreesFromBackwards = TO_DEG * acos(min(1., max(-1., direction->Unit().Dot(-velocity.Unit()))));
+            double turnFromBackwardsTime = degreesFromBackwards / ship.TurnRate();
+            forwardTime += turnFromBackwardsTime;
+			
+			// Time to turn from facing forwards to target:
+            double degreesFromForward = TO_DEG * acos(min(1., max(-1., direction->Unit().Dot(angle.Unit()))));
+            double turnFromForwardTime = degreesFromForward / ship.TurnRate();
+            reverseTime += turnFromForwardTime;
```

Comment:
Formatting issue: it looks like you're using spaces for indentation here rather than tabs.